File: PI_Calculator.cpp
#include <cmath>
#include <iostream>

int main()
{
    long long int num_of_iter = 1000000;
    double MY_PI = 0;
    double dx = 2.0/num_of_iter;
    for(long long int i = 0; i <= num_of_iter; i++)
    {
        MY_PI += dx * sqrt((-1 * ((-1 + dx * i) * (-1 + dx * i))) + 1);
        std::cout<<i<<std::endl;
        
    }
    MY_PI *= 2 ;
    std::cout<< MY_PI<<std::endl;
}

File: PI_Calculator.py
import math
num_of_iter = 1000000
MY_PI = 0
dx = 2.0/num_of_iter
for i in range(num_of_iter):#gets riemann sum of half circle
    MY_PI += dx * math.sqrt((-1 * ((-1 + dx * i) * (-1 + dx * i))) + 1) 
    print(i)    
MY_PI *= 2 #since we take riemann sum of half circle
print("PI Approximately Equals ")
print(MY_PI)
print("Margin of Error is +-")
print(2 * 2 * dx) # I eblieve this is how you calculate Margin of Error 2 = size of circle and 2 is due too pi being multiplied by 2
